### PR TITLE
Remove initializer requirement.

### PR DESCRIPTION
**Pull request by nashby**
Changes
- Use `Kernel#caller_locations` in `ViewComponent::Base::source_location` to get components location without requiring component to define `initialize` method.
- Remove initializer requirement from README

### CODE REVIEW

**Comment by joelhawksley** [comment]
@nashby excellent work! 

A couple things:
- Mind adding yourself to the README and updating the CHANGELOG? 
- Could you clean up the example components in `test` to remove any initializers that are unnecessary now? They look like `def initialize(*); end`.

**Comment by nashby** [comment]
@joelhawksley thanks for reviewing! All changes are done. I also update generator to not generate blank initialize method (with a test)

File: README.md [hunk 1]
```diff
@@ -498,10 +493,10 @@ Bug reports and pull requests are welcome on GitHub at https://github.com/github
 |@mellowfish|@horacio|@dukex|@dark-panda|@smashwilson|
 |Spring Hill, TN|Buenos Aires|São Paulo||Gambrills, MD|
 
-|<img src="https://avatars.githubusercontent.com/blakewilliams?s=256" alt="blakewilliams" width="128" />|<img src="https://avatars.githubusercontent.com/seanpdoyle?s=256" alt="seanpdoyle" width="128" />|<img src="https://avatars.githubusercontent.com/tclem?s=256" alt="tclem" width="128" />|
+|<img src="https://avatars.githubusercontent.com/blakewilliams?s=256" alt="blakewilliams" width="128" />|<img src="https://avatars.githubusercontent.com/seanpdoyle?s=256" alt="seanpdoyle" width="128" />|<img src="https://avatars.githubusercontent.com/tclem?s=256" alt="tclem" width="128" />|<img src="https://avatars.githubusercontent.com/nashby?s=256" alt="nashby" width="128" />
 |:---:|:---:|:---:|
-|@blakewilliams|@seanpdoyle|@tclem|
-|Boston, MA|New York, NY|San Francisco, CA|
+|@blakewilliams|@seanpdoyle|@tclem|@nashby|
+|Boston, MA|New York, NY|San Francisco, CA|Minsk
```
Comment (joelhawksley) [inline README.md]:
```suggestion
|Boston, MA|New York, NY|San Francisco, CA|Minsk|
```

File: README.md [hunk 2]
```diff
@@ -498,10 +493,10 @@ Bug reports and pull requests are welcome on GitHub at https://github.com/github
 |@mellowfish|@horacio|@dukex|@dark-panda|@smashwilson|
 |Spring Hill, TN|Buenos Aires|São Paulo||Gambrills, MD|
 
-|<img src="https://avatars.githubusercontent.com/blakewilliams?s=256" alt="blakewilliams" width="128" />|<img src="https://avatars.githubusercontent.com/seanpdoyle?s=256" alt="seanpdoyle" width="128" />|<img src="https://avatars.githubusercontent.com/tclem?s=256" alt="tclem" width="128" />|
+|<img src="https://avatars.githubusercontent.com/blakewilliams?s=256" alt="blakewilliams" width="128" />|<img src="https://avatars.githubusercontent.com/seanpdoyle?s=256" alt="seanpdoyle" width="128" />|<img src="https://avatars.githubusercontent.com/tclem?s=256" alt="tclem" width="128" />|<img src="https://avatars.githubusercontent.com/nashby?s=256" alt="nashby" width="128" />
 |:---:|:---:|:---:|
```
Comment (joelhawksley) [inline README.md]:
```suggestion
|:---:|:---:|:---:|:---:|
```

**Comment by nashby** [comment]
> ~One small change! (I'd make it myself, but can't commit to your branch)~ I made the change. Will merge when green.

Thanks!

**Pull request merged**